automotive/src/main/java/com/example/android/uamp/automotive/QrCodeSignInFragment.kt

```diff
@@ -39,34 +34,43 @@ import kotlinx.android.synthetic.main.qr_sign_in.toolbar
  */
 class QrCodeSignInFragment : Fragment(R.layout.qr_sign_in) {
 
+    private var binding: QrSignInBinding? = null
+
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
+        binding = QrSignInBinding.bind(view)
+
         val context = requireContext()
 
-        toolbar.setNavigationOnClickListener {
+        binding?.toolbar?.setNavigationOnClickListener {
             requireActivity().supportFragmentManager.popBackStack()
         }
 
-        app_icon.setImageDrawable(getDrawable(requireContext(), R.drawable.aural_logo))
-        primary_message.text = getString(R.string.qr_sign_in_primary_text)
-        secondary_message.text = getString(R.string.qr_sign_in_secondary_text)
+        binding?.appIcon?.setImageDrawable(getDrawable(requireContext(), R.drawable.aural_logo))
+        binding?.primaryMessage?.text = getString(R.string.qr_sign_in_primary_text)
+        binding?.secondaryMessage?.text = getString(R.string.qr_sign_in_secondary_text)
 
         // Links in footer text should be clickable.
-        footer.text = HtmlCompat.fromHtml(
+        binding?.footer?.text = HtmlCompat.fromHtml(
             context.getString(R.string.sign_in_footer),
             HtmlCompat.FROM_HTML_MODE_LEGACY
         )
-        footer.movementMethod = LinkMovementMethod.getInstance()
+        binding?.footer?.movementMethod = LinkMovementMethod.getInstance()
 
         setQrCode(getString(R.string.qr_code_url))
     }
 
+    override fun onDestroy() {
+        super.onDestroy()
+        binding = null
+    }
+
     /**
      * Sets the QR code rendered on this sign in screen.
      *
      * @param code The QR code to display.
      */
     private fun setQrCode(url: String) {
-        Glide.with(this).load(url).into(qr_code)
+        binding?.qrCode?.let { Glide.with(this).load(url).into(it) }
```

Comment:
Could you just pass the view in with the url (or, perhaps even better, just move it into `onViewCreated()`)?